File: BFS/boj_2606.py
import sys

input = sys.stdin.readline

n = int(input())

graph = [[0]*(n+1) for _ in range(n+1)]
visited = [0]*(n+1)

m = int(input())

for _ in range(m):
    a, b = map(int, input().split())
    graph[a][b] = graph[b][a] = 1

def bfs(V):
    q = [V]
    visited[V] = 1
    while q:
        V = q.pop(0)
        for i in range(1, n+1):
            if graph[V][i] == 1 and visited[i] == 0:
                q.append(i)
                visited[i] = 1

bfs(1)
cnt = 0
for i in visited:
    if i:
        cnt += 1

print(cnt-1)

File: Breadth-first Search/boj_2606.cpp
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

int bfs(int n, const vector<vector<int>>& graph) {
    int v, cnt = 0;
    queue<int> q;
    vector<int> visited(n + 1);
    
    q.push(1);
    visited[1] = 1;

    while (!q.empty()) {
        v = q.front();
        q.pop();
        cnt++;

        for (int nv : graph[v]) {
            if (!visited[nv]) {
                q.push(nv);
                visited[nv] = 1;
            }
        }
    }

    return cnt - 1;
}


int main() {
    int n, m;

    cin >> n;
    cin >> m;

    vector<vector<int>> graph(n + 1);

    while (m--) {
        int s, e;

        cin >> s >> e;

        graph[s].emplace_back(e);
        graph[e].emplace_back(s);
    }

    cout << bfs(n, graph);

    return 0;
}
